pkg/process/process_manager.go

```diff
@@ -104,6 +136,61 @@ func (pm *Manager) startMonitoring() {
 	}
 }
 
+func (pm *Manager) checkMountPointStatusForEngine() {
+	var processToUpdate []*Process
+	volumeMountPointMap, err := getVolumeMountPointMap()
+	if err != nil {
+		logrus.WithError(err).Warn("Failed to get all volume mount points")
+	}
+
+	pm.lock.RLock()
+	for _, p := range pm.processes {
+		p.lock.Lock()
+		if isEngineProcess(p) && p.State == StateRunning {
+
+			nameSlices := strings.Split(p.Name, "-")
+			volumeNameSHA := sha256.Sum256([]byte(strings.Join(nameSlices[:len(nameSlices)-2], "-")))
+
+			if mp, exists := volumeMountPointMap[hex.EncodeToString(volumeNameSHA[:])]; exists {
+				p.Conditions[types.EngineConditionFilesystemReadOnly] = util.IsMountPointReadOnly(mp)
+				processToUpdate = append(processToUpdate, p)
+			}
+		}
+		p.lock.Unlock()
```

Comment:
Move just next to `p.lock.Lock()`.

Reply:
we iterate every process
lock it and do some check and unlock it
if we put this next to lock
```
p.lock.Lock()
defer p.lock.Unlock()
```

Then every process will be locked in the end and need to wait until function return to unlock all the process
I think that will effect performance and might stuck some operation?

Reply:
yeah, misunderstood the scope of defer stack. No need to change here.